DigitalLearningSolutions.Data.Tests/DataServices/UserDataServiceTests/UserCentreDetailsDataServiceTests.cs

```diff
@@ -468,6 +473,34 @@ public void GetCentreEmailVerificationDetails_returns_expected_value()
             result.EmailVerificationHash.Should().Be(code);
             result.EmailVerified.Should().BeNull();
             result.EmailVerificationHashCreatedDate.Should().Be(createdDate);
+            result.CentreIdIfEmailIsForUnapprovedDelegate.Should().Be(null);
+        }
+
+        [Test]
+        public void GetCentreEmailVerificationDetails_returns_expected_value_for_centre_id_if_delegate_is_unapproved()
+        {
+            using var transaction = new TransactionScope();
+
+            // Given
+            const int userId = 1;
+            const int centreId = 2;
+            const string email = "unverified@email.com";
+            const string code = "code";
+            var createdDate = new DateTime(2022, 1, 1);
+
+            GivenEmailVerificationHashLinkedToUserCentreDetails(userId, centreId, email, code, createdDate);
+
+            // When
+            var result = userDataService.GetCentreEmailVerificationDetails(code);
+
+            // Then
+
+            result!.UserId.Should().Be(userId);
+            result.Email.Should().Be(email);
+            result.EmailVerificationHash.Should().Be(code);
+            result.EmailVerified.Should().BeNull();
+            result.EmailVerificationHashCreatedDate.Should().Be(createdDate);
+            result.CentreIdIfEmailIsForUnapprovedDelegate.Should().Be(centreId);

```

Comment:
It's unlikely that the delegateisApproved value is ever going to affect the other values, even if a dev makes a mistake, so you just have to check whether the approved value is correct.

Reply:
Sorry what do you mean? There's not a value for DelegateisApproved in the model that I can check, just the CentreIdIfEmailIsForUnapprovedDelegate, should I add one?

Reply:
Sorry, I meant you just have to check whether the centre id is correct.